cli/cli/kurtosis_gateway/connection/provider.go

```diff
@@ -211,7 +182,14 @@ func (provider *GatewayConnectionProvider) getRunningPodNamesByLabels(namespace
 	return runningPodNames, nil
 }
 
+// TODO - this function shouldn't exist when kt- + enclave name is the namespace inside kurtosis
+// https://github.com/kurtosis-tech/kurtosis/issues/1203 - till then we cache it
 func (provider *GatewayConnectionProvider) getEnclaveNamespaceNameForEnclaveId(enclaveId string) (string, error) {
+	enclaveNamespaceName, found := provider.enclaveIdToEnclaveNamespaceName[enclaveId]
```

Comment:
Great workaround until the namespace names are deterministic.

Reply:
Thank you! :-)